### PR TITLE
Image deprecations and optimisation

### PR DESCRIPTION
**Pull request by jonom**
Consolidates thumbnail methods to use the common image manipulation API, to prevent unnecessary creation of identical images with different filenames. Generate methods also tagged for removal in 4.0.

### CODE REVIEW

File: model/Image.php [hunk 1]
```diff
@@ -582,38 +592,77 @@ public function generateSetSize(Image_Backend $backend, $width, $height) {
...
 		return $backend->paddedResize($this->stat('cms_thumbnail_width'),$this->stat('cms_thumbnail_height'));
 	}
 
 	/**
 	 * Resize this image for preview in the Asset section. Use in templates with $AssetLibraryPreview.
+	 *
+	 * @return Image_Cached|null
+	 */
+	public function AssetLibraryPreview() {
+		return $backend->paddedResize($this->stat('asset_preview_width'),$this->stat('asset_preview_height'));
```
Comment (tractorcow) [inline model/Image.php]: Undefined variable $backend here. :)

Comment (jonom) [inline model/Image.php]: Oops!

**Comment by jonom** [comment]
Have reduced this a bit - AssetLibraryPreview and AssetLibraryThumbnail don't seem to be used anywhere in Framework or CMS and are not present in 4.0 so no sense introducing new API. I notice the thumbnail config variables are a bit of a mess and used in unintended ways but I see you've addressed that in 4.0 (Thanks!). So it's really just StripThumbnail getting it's own method now, and a few deprecation notices.

File: model/Image.php [hunk 2]
```diff
@@ -582,38 +592,59 @@ public function generateSetSize(Image_Backend $backend, $width, $height) {
...
+		Deprecation::notice('4.0', 'Generate methods are no longer applicable');
 		return $backend->paddedResize($this->stat('asset_thumbnail_width'),$this->stat('asset_thumbnail_height'));
 	}
 
 	/**
 	 * Resize this image for use as a thumbnail in a strip. Use in templates with $StripThumbnail.
+	 *
+	 * @return Image_Cached|null
+	 */
+	public function StripThumbnail() {
```
Comment (tractorcow) [inline model/Image.php]: Can you also PR to master to make sure this method is in ImageManipulations trait?

Comment (jonom) [inline model/Image.php]: It's [already there](http://api.silverstripe.org/master/class-SilverStripe.Filesystem.ImageManipulation.html#_StripThumbnail) ;)

Comment (tractorcow) [inline model/Image.php]: You're so efficient you'd done it before I'd even asked!

**Pull request merged**